### PR TITLE
COVID Add New Test - Add modals for describing test mechanism and antibody types

### PR DESCRIPTION
**Pull request by juliatan**
# Description

Notion: https://www.notion.so/joinzoe/COVID-Test-Screen-UI-update-Add-New-Test-Screen-a4fe50cf67eb44b79af00da3ace46652

Figma: https://www.figma.com/file/7TEcX7f6rw3Z0WYR4TpX3R/COVID-Symptom-Study-(App)---MASTER?node-id=12017%3A157720

**Note: I'm currently using the custom Modal we've used throughout our app. There's no easy way to enforce a stick button on it. Also, on Android, if it's a larger modal, there isn't any marginVertical set. I'm guessing this permeates throughout our app in terms of design. If we change this, we'd need to trawl through our other modals to ensure nothing is adversely affected I think.**

**Note: This is an example where the fonts are confusing. I've created some new Text blocks so we can refactor easier when we get round to it.**

## On what platform have you tested the change?

- [x] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Infomodals

androidinfomodal

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [x] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

File: src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx [hunk 1]
```diff
@@ -9,10 +10,13 @@ import {
   ECovidTestTrainedWorkerOptions,
 } from '@covid/core/user/dto/UserAPIContracts';
 import i18n from '@covid/locale/i18n';
+import { colors } from '@theme';
 import { FormikProps } from 'formik';
 import * as React from 'react';
 import * as Yup from 'yup';
 
+import { CovidTestAntibodyInfoModal } from '../components/CovidTestAntibodyInfoModal';
```
Comment (aikewoody) [inline src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx:18]: Try to only use absolute import statements

File: src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx [hunk 2]
```diff
@@ -88,11 +92,25 @@ export const CovidTestMechanismQuestion: ICovidTestMechanismQuestion<IProps, ICo
...
 
+  const [showMechanismModal, setShowMechanismModal] = React.useState(false);
+  const openMechanismModal = () => {
+    setShowMechanismModal(true);
+  };
+
+  const [showAntibodyModal, setShowAntibodyModal] = React.useState(false);
+  const openAntibodyModal = () => {
+    setShowAntibodyModal(true);
+  };
```
Comment (aikewoody) [inline src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx:103]: Why not also make a function for closing? e.g.
```
  const closeAntibodyModal = () => {
    setShowAntibodyModal(false);
  };
```

File: src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx [hunk 3]
```diff
@@ -88,11 +92,25 @@ export const CovidTestMechanismQuestion: ICovidTestMechanismQuestion<IProps, ICo
...
+
   return (
     <>
+      {showMechanismModal ? <CovidTestMechanismInfoModal onRequestClose={() => setShowMechanismModal(false)} /> : null}
+      {showAntibodyModal ? <CovidTestAntibodyInfoModal onRequestClose={() => setShowAntibodyModal(false)} /> : null}
```
Comment (aikewoody) [inline src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx:108]: Wouldn't it be easier to use the visible prop of the `Modal`?
`<CovidTestAntibodyInfoModal onRequestClose={closeAntibodyModal} visible={showAntibodyModal} />`

File: src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx [hunk 4]
```diff
@@ -124,6 +142,8 @@ export const CovidTestMechanismQuestion: ICovidTestMechanismQuestion<IProps, ICo
         <RadioInput
           required
           error={formikProps.touched.antibody ? formikProps.errors.antibody : ''}
+          IconComponent={() => <InfoCircle color={colors.primary} height={15} width={15} />}
```
Comment (aikewoody) [inline src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx:145]: Same component as above, better to define it once and then reuse

**Review by aikewoody — APPROVED** [review]
Great work!

File: src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx (hunk 2 above)
Comment (juliatan) [reply src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx:103]: My thinking is that the `setShowAntibodyModal` is already the closing function and seems redundant if we don't want other side effects. 🤷🏻‍♀️

File: src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx (hunk 3 above)
Comment (juliatan) [reply src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx:108]: Yes, good shout! 😄

File: src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx (hunk 4 above)
Comment (juliatan) [reply src/features/covid-tests/fields/CovidTestMechanismQuestion.tsx:145]: Ok yep, will do.

**Pull request merged**